mathesar_ui/src/components/moveable/Resizeable.svelte

```diff
@@ -0,0 +1,53 @@
+<script lang="ts">
+  import Moveable from 'svelte-moveable';
+  /**
+   * The HTML Element which is the movable target
+   * @required
+   */
+  export let target : HTMLElement = null;
+
+  /**
+   * Callback function when resize event is triggered
+   * This is wrapped in a function which sizes the target element.
+   */
+  export let onResize:CallableFunction = () => {};
+
+  /**
+   * Callback function when resize has ended
+   */
+  export let onResizeEnd:CallableFunction = () => {};
+  export let keepRatio = false;
+  export let padding = {
+    top: 0, right: 0, bottom: 0, left: 0,
+  };
+  export let directions = [
+    'nw', 'n', 'ne', 'w', 'e', 'sw', 's', 'se',
+  ];
+  export let throttle = 0;
+  export let zoom = 0.5;
+
+  function resize(event:CustomEvent) {
+    target.style.width = `${<number>event.detail.width}px`;
+    target.style.height = `${<number>event.detail.height}px`;
+    onResize(event);
+  }
+
+  function resizeEnd(event:CustomEvent) {
+    onResizeEnd(event);
+  }
+
+  export let moveable: Moveable = null;
+</script>
+
+<Moveable bind:this={moveable}
+  keepRatio={keepRatio}
+  on:resize={resize}
+  on:resizeEnd={resizeEnd}
+  origin={true}
+  padding={padding}
+  renderDirections={directions}
+  resizable={true}
+  target={target}
+  throttleResize={throttle}
+  zoom={zoom}
+/>
```

Comment:
@pegnott The svelte-movable package exports a component. For utilities that involve making changes on existing dom elements (such as popper, portal) , it would be best to use an action.

For more information on actions, please refer https://svelte.dev/docs#use_action. You could also look into popper.ts (within /components/common/actions) file in our code base, which is an action wrapper over [popper](https://popper.js.org/).

For our usecase, it would be best to install only the movable package `npm i moveable`, and create an action over it.

Reply:
> @pegnott The add column functionality looks good to me! Thanks!
> 
> Resizing columns do not seem stable. I have added a few review comments on it.

Thank you. I am addressing these this afternoon.